trusted-firmware-m/toolchain_GNUARM.cmake

```diff
@@ -13,8 +13,13 @@ endif()
 
 set(CMAKE_SYSTEM_NAME Generic)
 
-set(CMAKE_C_COMPILER ${CROSS_COMPILE}-gcc)
-set(CMAKE_ASM_COMPILER ${CROSS_COMPILE}-gcc)
+if (${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Windows")
+    set(CMAKE_C_COMPILER ${CROSS_COMPILE}-gcc.exe)
```

Comment:
This would be a more correct  way to fix this issue:
```
find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}-gcc)
find_program(CMAKE_ASM_COMPILER ${CROSS_COMPILE}-gcc)
```

then there is no need for testing platform, and it even would allow a caller to fixed the CMAKE_C_COMPILER in the cache, if need be.

Reply:
The CROSS_COMPILE var also contains the path. Will this still work or do we need to split it @tejlmand?

Reply:
no, that will work.
It just mean CMake won't look into other places for finding the program.
As example:
```
find_program(TEST /bin/test)
```
will not find candidates under `/usr/bin/test`.

Reply:
Ok, but that sounds like what we want. Should I add checks for `-NOTFOUND`?

Reply:
> Ok, but that sounds like what we want. Should I add checks for `-NOTFOUND` @tejlmand?

That is always a good thing to do as it can also detect wrong values being provided by `${CROSS_COMPILE}`, even though it wasn't verified before.

Reply:
Done